### DISCUSSION TITLE
RSS/JSON: Add title attribute to 'link rel="alternate"'

### DISCUSSION THREAD

**Discussion by bttrx** [GetPublii/Publii#1915] [Ideas]
Would be more comfortable for people when subscribing.

```
<link rel="alternate" type="application/atom+xml" href="http://SITE/feed.xml" />
<link rel="alternate" type="application/json" href="http://SITE/feed.json" />
```

```
<link rel="alternate" type="application/atom+xml" href="http://SITE/feed.xml" title="SITE NAME - Atom"/>
<link rel="alternate" type="application/json" href="http://SITE/feed.json" title="SITE NAME - JSON"/>
```

I think, this file needs to be changed: https://github.com/GetPublii/Publii/blob/master/app/back-end/modules/render-html/handlebars/helpers/feed-link.js

**Comment by michaelfluegge** [comment]
you can build a theme override and put your custom feed-link.js there:
https://getpublii.com/dev/theme-overrides/

**Reply by bttrx** [reply]
Thanks, but that didn't work for me.

Whatever I return in feedLink() doesn't make it to the final output files. I guess, my `input/themes/terminal-override/feed-link.js` is never called.

As a workaround I now use a customized `partials/head.hbs`:

```diff
--- head.hbs.bak	2024-12-19 15:23:42.026107000 +0100
+++ head.hbs	2025-01-05 21:54:06.187910035 +0100
@@ -11,11 +11,12 @@
         {{/if}}
         {{metaDescription}}
         {{metaRobots}} 
         {{{publiiHead}}}
         {{canonicalLink}}
-        {{feedLink}}
+        <link rel="alternate" type="application/atom+xml" href="{{@website.url}}feed.xml" title="{{@website.name}}" />
+        <link rel="alternate" type="application/json" href="{{@website.url}}feed.json" title="{{@website.name}}" />
         {{socialMetaTags}}
 		{{#if @config.custom.faviconUpload}}
 	        <link rel="shortcut icon" href="{{@config.custom.faviconUpload}}" type="{{@config.custom.faviconExtension}}" />
         {{/if}}
         {{#if @pagination}}
```

Of course, this a little unsafe, because it assumes both feeds being generated all the time. If I may turn this option one day...

So, this is still a feature request of mine to include the title attribute into a Publii release.

**Reply by candidexmedia** [reply]
> Thanks, but that didn't work for me.

@bttrx Haven't tested this, but I wonder if creating a new, custom helper would do the trick?

See: https://getpublii.com/dev/how-to-create-a-custom-helper/

**Reply by bttrx** [reply]
@candidexmedia Thanks, but my trick is enough for me until the title attribute makes it to a Publii release some day.

**Comment by dziudek** [comment]
@bttrx - in v.0.46.4 we will add `title` attributes to feed tags - thank you for this suggestion!
